packages/components/src/internal/actions.ts

```diff
@@ -107,6 +93,40 @@ export function selectGridIdsFromTransactionId(
     });
 }
 
+export async function selectGridIdsFromTransactionId(
+    gridIdPrefix: string,
+    schemaQuery: SchemaQuery,
+    transactionAuditId: number,
+    dataType: string,
+    actions: Actions
+): Promise<string[]> {
+    if (!transactionAuditId) {
+        return;
+    }
+    const modelId = createGridModelId(gridIdPrefix, schemaQuery);
+    const selected = await getGridIdsFromTransactionId(transactionAuditId, dataType);
+    await setSelected(modelId, true, selected, undefined, true, schemaQuery.schemaName, schemaQuery.queryName);
+    actions.replaceSelections(modelId, selected);
+    actions.loadModel(modelId, true);
+}
+
+export async function getSampleDataFromTransactionIds(transactionAuditId: number):Promise<{rowIds: string[], sampleTypes: string[]}> {
```

Comment:
I was a little confused at first by the "SampleData" part of this function name when I looked at the return type / info. I thought it was returning the data for the samples, but it is just the distinct set of sample types for those ids. I can't think of a different name though...maybe `getSampleTypesFromTransactionId`?

Reply:
Done.